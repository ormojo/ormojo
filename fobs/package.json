{
  "name": "@ormojo/fobs",
  "version": "1.0.7",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "description": "Fast, minimal, functional ES7 Observables.",
  "type": "commonjs",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "scripts": {
    "clean": "rimraf lib coverage .nyc_output",
    "build": "npm run build:rollup",
    "build:rollup": "rollup -c",
    "build:plain": "babel -x \".js\",\".lsc\" src --out-dir lib",
    "build:rollup:coverage": "cross-env COVERAGE=true rollup -c",
    "test": "npm run lint && npm run clean && npm run build && jest",
    "test:only": "jest",
    "coverage": "npm run clean && npm run build:rollup:coverage && jest --coverage",
    "lint": "eslint --ext .js,.lsc src",
    "preversion": "npm run test"
  },
  "dependencies": {
    "symbol-observable": "^1.1.0"
  },
  "devDependencies": {
    "jest": "^26.1.0",
    "npm-run-all": "^4.1.5"
  },
  "jest": {
    "testEnvironment": "node",
    "moduleFileExtensions": [
      "js",
      "json",
      "lsc"
    ],
    "transform": {
      "\\.lsc$": ["babel-jest", { "rootMode": "upward" }]
    },
    "testRegex": "(\\.|/)test\\.lsc$",
    "coveragePathIgnorePatterns": [
      "/node_modules/",
      "/lib/"
    ]
  }
}
