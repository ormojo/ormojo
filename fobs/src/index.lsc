export * from './util'

export { createBehaviorSubject } from './constructors/createBehaviorSubject'
export { createSubject } from './constructors/createSubject'
export { toBehaviorSubject } from './constructors/toBehaviorSubject'
export { createConstant } from './constructors/createConstant'

export { changed } from './operators/changed'
export { combineLatest } from './operators/combineLatest'
export { filter } from './operators/filter'
export { map } from './operators/map'
export { memoize } from './operators/memoize'
export { merge } from './operators/merge'
export { switchLatest } from './operators/switchLatest'
