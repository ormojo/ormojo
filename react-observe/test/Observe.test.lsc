import React from 'react'
import renderer from 'react-test-renderer'
import { createBehaviorSubject } from '@ormojo/fobs'
import { Observe } from '..'

import { RenderCounter } from './RenderCounter'

test! 'works', ->
  let tree
  o = createBehaviorSubject()
  nonce = {x : 1}
  o.next(nonce)

  component = renderer.create(
    <Observe source={o}>{o ->
      <RenderCounter value={JSON.stringify(o)} />
    }</Observe>
  )

  now tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  o.next(nonce)
  now tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  o.next({x: 1})
  now tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  o.next({x: 2})
  now tree = component.toJSON()
  expect(tree).toMatchSnapshot()
