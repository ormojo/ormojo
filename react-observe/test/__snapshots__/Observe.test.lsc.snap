// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
<div>
  ["{\\"x\\":1}"]
  {"x":1}
</div>
`;

exports[`works 2`] = `
<div>
  ["{\\"x\\":1}"]
  {"x":1}
</div>
`;

exports[`works 3`] = `
<div>
  ["{\\"x\\":1}","{\\"x\\":1}"]
  {"x":1}
</div>
`;

exports[`works 4`] = `
<div>
  ["{\\"x\\":1}","{\\"x\\":1}","{\\"x\\":2}"]
  {"x":2}
</div>
`;
